managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -324,24 +326,25 @@ public Map<String, String> getCursorProperties() {
         return cursorProperties;
     }
 
-    @Override
-    public CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties) {
+    private CompletableFuture<Void> asyncReadAndUpdateCursorProperties(
+            final Function<Map<String, String>, Map<String, String>> updateFunction) {
         CompletableFuture<Void> updateCursorPropertiesResult = new CompletableFuture<>();
         ledger.getStore().asyncGetCursorInfo(ledger.getName(), name, new MetaStoreCallback<>() {
             @Override
             public void operationComplete(ManagedCursorInfo info, Stat stat) {
+                final Map<String, String> newProperties = updateFunction.apply(ManagedCursorImpl.this.cursorProperties);
```

Comment:
Here might introduce a problem, the `ManagedCursorImpl.this.cursorProperties` is staled, but we will update the zookeeper with a new stat.

Reply:
Update the zookeeper with local stat?

Reply:
We should use the properties from `info`.

Reply:
@Jason918 Maybe we can be removed `asyncGetCursorInfo` and cache `ManagedCursorInfo`, What do you think?